build.sh

```diff
@@ -16,13 +16,140 @@
 
```

Comment:
I will definitely keep that in mind if it gets any bigger, or stops having a straight-line control flow.

I should confess that I've written 10,000 line shell scripts before, so my tolerance for them is probably higher than other peoples' :)